### PR TITLE
MGDAPI-1882 - Update/Create automated e2e tests/test cases - DO NOT Merge. Deprecated

### PR DESCRIPTION
**Pull request by valerymo**
This PR is deprecated. New PR based on master created: https://github.com/integr8ly/integreatly-operator/pull/2188
# Issue link
https://issues.redhat.com/browse/MGDAPI-1882

# What
Automation tests update for Observability Operator.
AMO Automation tests - continue work (will be removed in different Task/PR). Some AMO Automation tests  code was impacted, to allow similar tests both for AMO and Observability Operator meanwhile.

# Verification steps
- Check that new Automation Tests (named as E00**) are running (command line single tests)
- Check that AMO  Automation Tests continue run successfully  (command line single tests)
- New automated test case running in pipeline without errors
- AMO automated test case running in pipeline without errors

### CODE REVIEW

File: test/common/alerts_exist.go [hunk 1]
```diff
@@ -895,6 +901,18 @@ func managedApiAwsExpectedRules(installtionName string) []alertsTestRule {
...
 
 func TestIntegreatlyAlertsExist(t TestingTB, ctx *TestingContext) {
+	prometheusPod := "prometheus-application-monitoring-0"
+	execToPodCommand := "curl localhost:9090/api/v1/rules"
+	testIntegreatlyAlertsExist(t, ctx, MonitoringOperatorNamespace, prometheusPod, execToPodCommand)
+}
+
+func TestIntegreatlyAlertsExistObservability(t TestingTB, ctx *TestingContext) {
+	prometheusPod := "prometheus-kafka-prometheus-0"
+	execToPodCommand := "wget -qO - localhost:9090/api/v1/rules"
```
Comment (makslion) [inline test/common/alerts_exist.go:911]: just out of curiosity - is there a reason to have `curl` and `wget` there? I mean why both and not any particular? :thinking:

Comment (valerymo) [reply test/common/alerts_exist.go:911]: Hi Maksym. The reason is that prometheus-kafka-prometheus-0 pod in OO is missing curl

**Comment by laurafitzgerald** [comment]
Hi @valerymo can you drop the commit `update vendor for OO v3.0.4` from this pr. It got added when we did the rebase. I also had the same problem on another pr.

**Comment by valerymo** [comment]
Thanks very much Laura, commit update vendor for OO v3.0.4 - removed.

**Comment by makslion** [comment]
Confirmed that `C04` passes locally after the recent push.

File: test/common/alerts_exist.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ var (
 	fileAdditional alertsTestFileStatus = "File found but not expected"
 	fileExists     alertsTestFileStatus = "File found with missing or unexpected rules"
 	fileCorrect    alertsTestFileStatus = "File found with all alerts present"
+	NamespacePrefixObservability = "redhat-rhoam-observability-"
```
Comment (MStokluska) [inline test/common/alerts_exist.go:38]: @valerymo would it be better to follow same approach as we currently do and push the namespace to test/common/types - where we could use existing prefix + observability and call it directly from there?

File: test/common/dashboards_data.go [hunk 3]
```diff
@@ -78,15 +78,20 @@ var (
...
 )
 
+func TestDashboardsDataObservability(t TestingTB, ctx *TestingContext){
+	execToPodCommand := "wget -qO - "
+	testDashboardsData(t, ctx, ObservabilityProductNamespace, execToPodCommand)
+}
+
 // TestDashboardsData verifies that all dashboards are installed and all the graphs are filled with data
-func TestDashboardsData(t TestingTB, ctx *TestingContext) {
+func testDashboardsData(t TestingTB, ctx *TestingContext, monitoringNamespace string, execToPodCommand string) {
```
Comment (MStokluska) [inline test/common/dashboards_data.go:87]: same question about monitoring namespace as in above file

File: test/common/namespace_restoration.go [hunk 4]
```diff
@@ -319,6 +329,31 @@ func removeMonitoringFinalizers(ctx *TestingContext, nameSpace string) error {
...
+	err = removAppObservabilityFinalizers(ctx, nameSpace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func removAppObservabilityFinalizers(ctx *TestingContext, nameSpace string) error {
+	//TODO
```
Comment (MStokluska) [inline test/common/namespace_restoration.go:348]: Just a question - this is going to stay as it is for now? (just checking!)

File: test/common/alerts_firing.go [hunk 5]
```diff
@@ -201,15 +208,22 @@ func getPodNamespaces(installType string) []string {
...
+func TestIntegreatlyAlertsPendingOrFiringObservability(t TestingTB, ctx *TestingContext) {
+	prometheusPod := "prometheus-kafka-prometheus-0"
+	execToPodCommand := "wget -qO - localhost:9090/api/v1/alerts"
+	testIntegreatlyAlertsPendingOrFiring(t, ctx, ObservabilityProductNamespace, prometheusPod, execToPodCommand)
+}
+
 // TestIntegreatlyAlertsFiring reports any firing or pending alerts
-func TestIntegreatlyAlertsPendingOrFiring(t TestingTB, ctx *TestingContext) {
+func testIntegreatlyAlertsPendingOrFiring(t TestingTB, ctx *TestingContext, monitoringNamespace string,
+											prometheusPod string,	command string) {
```
[the remark this answers is not in the document]
Comment (valerymo) [reply test/common/alerts_firing.go:219]: Done in new PR https://issues.redhat.com/browse/MGDAPI-1882

File: test/common/dashboards_data.go (hunk 3 above)
Comment (valerymo) [reply test/common/dashboards_data.go:87]: Done in new PR: https://github.com/integr8ly/integreatly-operator/pull/2188

File: test/common/routes_exist.go [hunk 6]
```diff
@@ -162,7 +162,7 @@ var rhmi2ExpectedRoutes = map[string][]ExpectedRoute{
 	"amq-online":                     amqOnlineRoutes,
 	"codeready-workspaces":           codeReadyRoutes,
 	"fuse":                           fuseRoutes,
-	"middleware-monitoring-operator": middlewareMonitoringRoutes,
+	"observability": 				  observabilityRoutes,
```
[the remark this answers is not in the document]
Comment (valerymo) [reply test/common/routes_exist.go:165]: Done in new PR: https://github.com/integr8ly/integreatly-operator/pull/2188

**Comment by valerymo** [comment]
old, depricated

**Pull request closed without merging**